src/math/lut_trig.c

```diff
@@ -0,0 +1,574 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2016-2023 Intel Corporation. All rights reserved.
+//
+// Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+
+#include <sof/audio/format.h>
+#include <sof/math/lut_trig.h>
+#include <stdint.h>
+
+#define SOFM_LUT_SINE_C_Q20 341782638 /* 2*SINE_NQUART/pi in Q12.20 */
+#define SOFM_LUT_SINE_NQUART 512 /* Must be 2^N */
+#define SOFM_LUT_SINE_SIZE (SOFM_LUT_SINE_NQUART + 1)
+
+/* An 1/4 period of sine wave as Q1.31 */
+const int32_t sofm_lut_sine_table[SOFM_LUT_SINE_SIZE] = {
```

Comment:
TODO: Should test if this could be int16_t and size 257 for 16 bit sine.

Reply:
one is check wether can use 16-bit， another thing is whether can shorten the size? do we really need 512? 
such a small usage cost 2k bytes table, it is big cost.

Also, could you add comments for how this table calculated? sin(i/2pi)? then we can further figure out whether 512 is must.

Reply:
static ?